packages/appium/test/cli-e2e-specs.js

```diff
@@ -146,4 +146,81 @@ describe('Driver CLI', function () {
       await fs.exists(installPath).should.eventually.be.false;
     });
   });
+
+  describe('run', function () {
+    before(async function () {
+      await clear();
+      await run('install', [localFakeDriverPath, '--source', 'local', '--json']);
+    });
+    it('should run a valid driver, valid script, and result in success', async function () {
+      const driverName = 'fake';
+      const scriptName = 'fake-success';
+      const out = JSON.parse(await run('run', [driverName, scriptName, '--json']));
+      out.should.not.include.key('error');
+    });
+    it('should run a valid driver, valid error prone script, and return error in json', async function () {
+      const driverName = 'fake';
+      const out = JSON.parse(await run('run', [driverName, 'fake-error', '--json']));
+      out.should.include.key('error');
+    });
+    it('should take a valid driver, invalid script, and throw an error', async function () {
+      const driverName = 'fake';
+      await chai.expect(run('run', [driverName, 'foo', '--json'])).to.eventually.be.rejectedWith(Error);
+    });
+    it('should take an invalid driver, invalid script, and throw an error', async function () {
+      const driverName = 'foo';
+      await chai.expect(run('run', [driverName, 'bar', '--json'])).to.eventually.be.rejectedWith(Error);
+    });
+  });
+});
+
+describe('Plugin CLI', function () {
+  let appiumHome;
+  const fakePluginDir = path.resolve(cwd, 'node_modules', '@appium', 'fake-plugin');
+  before(async function () {
```

Comment:
we duplicate all this before, after, run, and clear logic. seems like we could easily make them shared helper functions? (and there's probably no downside to sharing `appiumHome` between the driver and plugin tests here.)